moonraker/components/td1.py

```diff
@@ -0,0 +1,316 @@
+# Support for interfacing with TD-1 devices to assist in filament changers grabbing TD
+# and color for all filament loaded.
+#
+# Copyright (C) 2025 AJAX3D and Jim Madill <jcmadill1@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license
+
+import asyncio
+import serial
+import serial.tools.list_ports
+import logging
+import datetime
+
+from ..common import RequestType
+from ..common import WebRequest
+from ..utils import async_serial
+
+class TD1Protocol(asyncio.Protocol):
+    def __init__(self, config, serial_number, port, logger):
+        self.config = config
+        self.server = config.get_server()
+        self.serial_number = serial_number
+        # Setting port in config file
+        self.config.source.config.set('td1', 'serial', port)
+        self.logger = logger
+        self.buffer = ""
+        self.error_state = False
+        self.serial = async_serial.AsyncSerialConnection(config,
+                                                         config.get("baudrate", 115200))
```

Comment:
It is no longer necessary to pass a config object to AsyncSerial connection.  It will need a server object, a unique name, the port and the baudrate.